### PR TITLE
[BACKLOG-5408] - tests for kettle5-log4j-plugin

### PR DESCRIPTION
**Pull request by AndreyBurikhin**
@mdamour1976 please find someone who can review this. 
I marked some classes as deprecated in separate commit - they have negative effect on code coverage but I see no reason to waste time on them.

### CODE REVIEW

**Comment by dkincade** [comment]
@brosander @mattyb149

**Comment by mattyb149** [comment]
For all the added "deprecated" annotations, we need to indicate why they're deprecated and what should be used instead:

http://wiki.pentaho.com/display/PEOpen/Deprecation+Guidelines

**Comment by dkincade** [comment]
Excellent comment Matt - thanks!

**Comment by brosander** [comment]
@AndreyBurikhin seeing 2 failures when running here, looks like it could be timezone related

formatLogMessage_with_add_time
formatNonLogMessage_with_add_time

**Comment by AndreyBurikhin** [comment]
@brosander fix timezone issue

**Comment by AndreyBurikhin** [comment]
@mattyb149  I removed commit with "deprecated" annotations and created this case http://jira.pentaho.com/browse/BACKLOG-5447.

File: plugins/kettle5-log4j-plugin/ivy.xml [hunk 1]
```diff
@@ -25,7 +25,9 @@
 
     <dependency org="log4j" name="log4j" rev="1.2.17" conf="default->default" transitive="false"/>
 
-    <dependency org="junit" name="junit" rev="4.7" conf="test->default"/> 
+    <!-- test dependencies -->
+    <dependency org="junit" name="junit" rev="4.11" conf="test->default"/> 
+    <dependency org="org.mockito" name="mockito-all" rev="1.9.5" conf="test->default" transitive="false" />
```
Comment (mattyb149) [inline plugins/kettle5-log4j-plugin/ivy.xml]: I see other modules are using mockito 1.9.5 as well, though there was never a COMP case to cover it. I added one:

http://jira.pentaho.com/browse/COMP-2274

File: plugins/kettle5-log4j-plugin/src/org/pentaho/di/core/logging/log4j/Log4jLogging.java [hunk 2]
```diff
@@ -77,11 +73,20 @@ public void init() {
...
+
+  private void applyLog4jConfiguration() {
     LogLog.setQuietMode( true );
     LogManager.resetConfiguration();
     LogLog.setQuietMode( false );
-    DOMConfigurator.configure( PLUGIN_PROPERTIES_FILE );
+    DOMConfigurator.configure( getConfigurationFileName() );
+  }
+  
+  Logger createLogger( String loggerName ) {
```
Comment (mattyb149) [inline plugins/kettle5-log4j-plugin/src/org/pentaho/di/core/logging/log4j/Log4jLogging.java]: Package-protected methods need Javadoc to explain why they're not explicitly "protected".

File: plugins/kettle5-log4j-plugin/ivy.xml (hunk 1 above)
Comment (AndreyBurikhin) [inline plugins/kettle5-log4j-plugin/ivy.xml]: @mattyb149  @akhayrutdinov said that Joe rejecting COMP for testing artifacts e.g. http://jira.pentaho.com/browse/COMP-1412,
http://jira.pentaho.com/browse/COMP-1413.

File: plugins/kettle5-log4j-plugin/src/org/pentaho/di/core/logging/log4j/Log4jLogging.java (hunk 2 above)
Comment (AndreyBurikhin) [inline plugins/kettle5-log4j-plugin/src/org/pentaho/di/core/logging/log4j/Log4jLogging.java]: @mattyb149 done

**Comment by brosander** [comment]
Changed files:
==============
  plugins/kettle5-log4j-plugin/ivy.xml
  plugins/kettle5-log4j-plugin/src/org/pentaho/di/core/logging/log4j/Log4jLogging.java
  plugins/kettle5-log4j-plugin/test-src/.gitignore
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jFileAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jKettleLayoutTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jPipedAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/log4j/Log4jLoggingTest.java
  plugins/kettle5-log4j-plugin/test/.gitignore

Checkstyle violations:
----------------------
Errors:File plugins/kettle5-log4j-plugin/src/org/pentaho/di/core/logging/log4j/Log4jLogging.java1. Message: Line has trailing spaces. at Line: 38 Column: None2. Message: Line has trailing spaces. at Line: 40 Column: None3. Message: Line has trailing spaces. at Line: 83 Column: None




Unit test coverage change:
============
org.pentaho.di.core.logging.log4j.Log4jLogging.new Object() {...}:
 * linePercentCoverage :100.00%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :100.00%
 * instructionPercentCoverage :88.46%
    
org.pentaho.di.core.logging.Log4jStringAppender:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.CentralLogStore.new Object() {...}:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.LogWriter.new Object() {...}:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.Log4jKettleLayout.new ThreadLocal() {...}:
 * linePercentCoverage :100.00%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :100.00%
 * instructionPercentCoverage :100.00%
    
org.pentaho.di.core.logging.CentralLogStore:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.KettleLogChannelAppender:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.log4j.Log4jLogging:
 * branchPercentCoverage :100.00%
 * linePercentCoverage :65.22%
 * methodPercentCoverage :57.14%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :74.60%
    
org.pentaho.di.core.logging.Log4jKettleLayout:
 * branchPercentCoverage :45.45%
 * linePercentCoverage :45.83%
 * methodPercentCoverage :37.50%
 * complexityPercentCoverage :31.58%
 * instructionPercentCoverage :49.22%
    
org.pentaho.di.core.logging.Log4jPipedAppender:
 * linePercentCoverage :60.61%
 * methodPercentCoverage :56.25%
 * complexityPercentCoverage :56.25%
 * instructionPercentCoverage :55.67%
    
org.pentaho.di.core.logging.Log4jFileAppender:
 * linePercentCoverage :43.90%
 * methodPercentCoverage :36.84%
 * complexityPercentCoverage :36.84%
 * instructionPercentCoverage :38.93%
    
org.pentaho.di.core.logging.Log4jConsoleAppender:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.LogWriter:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%

**Comment by brosander** [comment]
Changed files:
==============
  plugins/kettle5-log4j-plugin/ivy.xml
  plugins/kettle5-log4j-plugin/src/org/pentaho/di/core/logging/log4j/Log4jLogging.java
  plugins/kettle5-log4j-plugin/test-src/.gitignore
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jFileAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jKettleLayoutTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jPipedAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/log4j/Log4jLoggingTest.java
  plugins/kettle5-log4j-plugin/test/.gitignore

Checkstyle violations:
----------------------





Unit test coverage change:
============
org.pentaho.di.core.logging.log4j.Log4jLogging.new Object() {...}:
 * linePercentCoverage :100.00%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :100.00%
 * instructionPercentCoverage :88.46%
    
org.pentaho.di.core.logging.Log4jStringAppender:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.CentralLogStore.new Object() {...}:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.LogWriter.new Object() {...}:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.Log4jKettleLayout.new ThreadLocal() {...}:
 * linePercentCoverage :100.00%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :100.00%
 * instructionPercentCoverage :100.00%
    
org.pentaho.di.core.logging.CentralLogStore:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.KettleLogChannelAppender:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.log4j.Log4jLogging:
 * branchPercentCoverage :100.00%
 * linePercentCoverage :65.22%
 * methodPercentCoverage :57.14%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :74.60%
    
org.pentaho.di.core.logging.Log4jKettleLayout:
 * branchPercentCoverage :45.45%
 * linePercentCoverage :45.83%
 * methodPercentCoverage :37.50%
 * complexityPercentCoverage :31.58%
 * instructionPercentCoverage :49.22%
    
org.pentaho.di.core.logging.Log4jPipedAppender:
 * linePercentCoverage :60.61%
 * methodPercentCoverage :56.25%
 * complexityPercentCoverage :56.25%
 * instructionPercentCoverage :55.67%
    
org.pentaho.di.core.logging.Log4jFileAppender:
 * linePercentCoverage :43.90%
 * methodPercentCoverage :36.84%
 * complexityPercentCoverage :36.84%
 * instructionPercentCoverage :38.93%
    
org.pentaho.di.core.logging.Log4jConsoleAppender:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.LogWriter:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%

**Comment by brosander** [comment]
Changed files:
==============
  plugins/kettle5-log4j-plugin/ivy.xml
  plugins/kettle5-log4j-plugin/src/org/pentaho/di/core/logging/log4j/Log4jLogging.java
  plugins/kettle5-log4j-plugin/test-src/.gitignore
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jFileAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jKettleLayoutTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jPipedAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/log4j/Log4jLoggingTest.java
  plugins/kettle5-log4j-plugin/test/.gitignore

Checkstyle violations:
----------------------
Errors:File plugins/kettle5-log4j-plugin/src/org/pentaho/di/core/logging/log4j/Log4jLogging.java
1. Message: Line has trailing spaces. at Line: 38 Column: None
2. Message: Line has trailing spaces. at Line: 40 Column: None
3. Message: Line has trailing spaces. at Line: 83 Column: None




Unit test coverage change:
============
org.pentaho.di.core.logging.log4j.Log4jLogging.new Object() {...}:
 * linePercentCoverage :100.00%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :100.00%
 * instructionPercentCoverage :88.46%
    
org.pentaho.di.core.logging.Log4jStringAppender:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.CentralLogStore.new Object() {...}:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.LogWriter.new Object() {...}:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.Log4jKettleLayout.new ThreadLocal() {...}:
 * linePercentCoverage :100.00%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :100.00%
 * instructionPercentCoverage :100.00%
    
org.pentaho.di.core.logging.CentralLogStore:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.KettleLogChannelAppender:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.log4j.Log4jLogging:
 * branchPercentCoverage :100.00%
 * linePercentCoverage :65.22%
 * methodPercentCoverage :57.14%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :74.60%
    
org.pentaho.di.core.logging.Log4jKettleLayout:
 * branchPercentCoverage :45.45%
 * linePercentCoverage :45.83%
 * methodPercentCoverage :37.50%
 * complexityPercentCoverage :31.58%
 * instructionPercentCoverage :49.22%
    
org.pentaho.di.core.logging.Log4jPipedAppender:
 * linePercentCoverage :60.61%
 * methodPercentCoverage :56.25%
 * complexityPercentCoverage :56.25%
 * instructionPercentCoverage :55.67%
    
org.pentaho.di.core.logging.Log4jFileAppender:
 * linePercentCoverage :43.90%
 * methodPercentCoverage :36.84%
 * complexityPercentCoverage :36.84%
 * instructionPercentCoverage :38.93%
    
org.pentaho.di.core.logging.Log4jConsoleAppender:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.LogWriter:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%

**Comment by AndreyBurikhin** [comment]
@brosander checkstyle violations were fixed

**Comment by mdamour1976** [comment]
@brosander Are you satisfied with the commit now?

**Comment by brosander** [comment]
@mdamour1976 I'll take another look

/cc @AndreyBurikhin

**Comment by brosander** [comment]
Changed files:
==============
  plugins/kettle5-log4j-plugin/ivy.xml
  plugins/kettle5-log4j-plugin/src/org/pentaho/di/core/logging/log4j/Log4jLogging.java
  plugins/kettle5-log4j-plugin/test-src/.gitignore
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jFileAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jKettleLayoutTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/Log4jPipedAppenderTest.java
  plugins/kettle5-log4j-plugin/test-src/org/pentaho/di/core/logging/log4j/Log4jLoggingTest.java
  plugins/kettle5-log4j-plugin/test/.gitignore

Checkstyle violations:
----------------------
No errors found




Unit test coverage change:
============
org.pentaho.di.core.logging.log4j.Log4jLogging.new Object() {...}:
 * linePercentCoverage :100.00%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :100.00%
 * instructionPercentCoverage :88.46%
    
org.pentaho.di.core.logging.Log4jStringAppender:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.CentralLogStore.new Object() {...}:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.LogWriter.new Object() {...}:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.Log4jKettleLayout.new ThreadLocal() {...}:
 * linePercentCoverage :100.00%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :100.00%
 * instructionPercentCoverage :100.00%
    
org.pentaho.di.core.logging.CentralLogStore:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.KettleLogChannelAppender:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.log4j.Log4jLogging:
 * branchPercentCoverage :100.00%
 * linePercentCoverage :65.22%
 * methodPercentCoverage :57.14%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :74.60%
    
org.pentaho.di.core.logging.Log4jKettleLayout:
 * branchPercentCoverage :45.45%
 * linePercentCoverage :45.83%
 * methodPercentCoverage :37.50%
 * complexityPercentCoverage :31.58%
 * instructionPercentCoverage :49.22%
    
org.pentaho.di.core.logging.Log4jPipedAppender:
 * linePercentCoverage :60.61%
 * methodPercentCoverage :56.25%
 * complexityPercentCoverage :56.25%
 * instructionPercentCoverage :55.67%
    
org.pentaho.di.core.logging.Log4jFileAppender:
 * linePercentCoverage :43.90%
 * methodPercentCoverage :36.84%
 * complexityPercentCoverage :36.84%
 * instructionPercentCoverage :38.93%
    
org.pentaho.di.core.logging.Log4jConsoleAppender:
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%
    
org.pentaho.di.core.logging.LogWriter:
 * branchPercentCoverage :0.00%
 * linePercentCoverage :0.00%
 * methodPercentCoverage :0.00%
 * complexityPercentCoverage :0.00%
 * instructionPercentCoverage :0.00%

**Pull request merged**